ToolkitApp/tabs/src/components/sample/ConnectedResources.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { Button } from '@fluentui/react-northstar'
+import { ClipboardCopiedToIcon } from '@fluentui/react-icons-northstar'
+import * as microsoftTeams from "@microsoft/teams-js"; 
+import "./ConnectedResources.css";
+import {TEAMS_CHANNEL_ID, CHAT, CHAT_ID} from './TabConstants';
+
+export function processTeamsContext(setResourceList,resourceList,setTitle){
+    //Get the context of where the tab is currently
+    microsoftTeams.getContext(function(context) {
+        
+        //Check if this is a teams channel
+        if(context.channelId && context.channelId.length !== 0){
+            if(isAlreadyPresent(resourceList, context.channelId)){
+                return; 
+            }
+            var channelId = createItemWithCopy(context.channelId);
+            channelId.header = TEAMS_CHANNEL_ID + context.channelId;
+            setTitle(context.teamName + " > " + context.channelName);
+            setResourceList([channelId])
+        }
+        //Check if it is a chat
+        else if(context.chatId && context.chatId.length !== 0){
+            if(isAlreadyPresent(resourceList, context.chatId)){
+                return; 
+            }
+            setTitle(CHAT);
+            var chatId = createItemWithCopy(context.chatId);
+            chatId.header = CHAT_ID + context.chatId;
+            setResourceList([chatId]);
+        }
+        else{
+            setTitle("");
+            setResourceList([]);
+        }
+    });
+}
+
+function isAlreadyPresent(list, id){
+    //if there aren't any contexts
+    if(list.length === 0){
+        return false;
+    }
+
+    //Loop through all the instances 
+    for(let i  = 0; i < list.length; i++){
+        if (list[i].header && list[i].header.includes(id)){
+            return true;
+        }
+    }
+    return false;
+}
+
+function copyText(id){
+    //Goal is to be able to have the user paste "id" once they try to copy 
+    var copyValue = document.createElement('textarea');
+    copyValue.value = id;
+    document.body.appendChild(copyValue);
+    copyValue.select();
+    document.execCommand("copy"); 
```

Comment:
This is directly modifying the DOM. Is there a way to not use `document`? In react it's usually bad to directly modify the DOM.

https://stackoverflow.com/questions/66620392/how-bad-is-it-to-change-the-dom-in-react

`In React, changing the DOM directly is usually bad because the state of the page should come directly from the state in React.`

Reply:
https://github.com/instructure/canvas-lms/issues/1767 Due to lack of compatibility with chrome, this was the alternative. It seems the workaround without using the DOM is to just use a library. I've done that in the last commit c0399e93467c354cf1c334cfcd9fa9c0a6270d56